site/src/main/java/uk/nhs/digital/ps/beans/Publication.java

```diff
@@ -236,6 +237,11 @@ public boolean isPubliclyAccessible() {
         return getChildBeansIfPermitted(PropertyKeys.ATTACHMENTS, HippoResourceBean.class);
     }
 
+    public List<Element> getSummaryParagraphs() {
+        assertPropertyPermitted(PropertyKeys.SUMMARY);
+        return getParsedText(PropertyKeys.SUMMARY);
+    }
```

Comment:
Publication now has two getters that return the same content in two different formats - `getSummary` and
`getSummaryParagraphs`, each returning the same content in two different formats. `BaseDocument` has gained `getParsedText` which isn't even needed by all of the extending classes. Both these additions mean that `Publication` (and other classes extending `BaseDocument`), apart from being concerned with just the content are now also burdened with formatting concerns which normally are of interest to the presentation/view layer rather than domain layer.

In recognition that it's difficult to completely divorce content of summary from _all_ formatting concerns (after all, we do want to let the users apply paragraphs and, later, possibly list of bullet points, and we do have to persist the information of this structure somehow), how does it sound if we:
1) introduced a class `StructuredText` to encapsulate collection of paragraphs,
1) made `Publication.getSummary` return instance of that class (the same for other classes extending `BaseDocument`),
1) moved `Publication.getSummaryParagraphs` to `StructuredText`, renaming it to something like `getElements`,
  `getComponents` or `getParts`,
1) moved `BaseDocument.parseText` to `StructuredText`.

This way, all low level manipulations related to text of this particular text format would be encapsulated in a class whose only concerns would be just that - to hold and correctly process a structured text rather than be concerned what's in it. Conversely, all that `Publication` would know is that it holds and returns a Summary - without being burdened with details of how to process it.

`StructuredText` could hold the raw text and parse it whenever its getters are called, or parse it once, when it's being created, and store collection of Elements. It could also expose `getFirstNCharacters(int charactersCount)` for where the summary needs to be displayed truncated in `publications-overview.ftl` and in `searchresults.ftl` (if/when we later add support for more types of elements, say lists of bullet points, this will allow to intelligently encapsulate logic for correct extraction of first characters - for example, it'll help dealing with the case when the first N characters spill beyond the first paragraph and into the list of bullet points).